mkosi

```diff
@@ -457,6 +457,34 @@ def install_arch(args, workspace, run_build_script):
 
     subprocess.run(cmdline, check=True)
 
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-networkd"], check=True)
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-resolved"], check=True)
+    os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
+    os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
+
+    nsswitch_old = os.path.join(workspace, "root", "etc/nsswitch.conf")
+    nsswitch_new = os.path.join(workspace, "root", "etc/nsswitch.conf.new")
+
+    with open(nsswitch_old, "r") as old:
+        with open(nsswitch_new, "w") as new:
+            for line in old:
+                if line.startswith("hosts:"):
+                    new.write(" ".join(["resolve" if x == "dns" else x for x in line.split(" ")]))
+                else:
+                    new.write(line)
+
+    shutil.copystat(nsswitch_old, nsswitch_new)
+    os.remove(nsswitch_old)
+    shutil.move(nsswitch_new, nsswitch_old)
```

Comment:
I’ve moved inside an `enable_networkd(…)` function everything which is related to networkd activation on ArchLinux:
* Enable `systemd-networkd` and `systemd-resolved` units;
* Replace `dns` by `resolve` in `nsswitch.conf`;
* Symlink `/etc/resolv.conf`.

I don’t know if it would make sense to split this function further because not all steps are needed on all distributions.